source/Symbol/SwiftASTContext.cpp

```diff
@@ -99,34 +99,32 @@
 #include "Plugins/SymbolFile/DWARF/DWARFASTParserSwift.h"
 
 #ifdef LLDB_CONFIGURATION_DEBUG
-#define VALID_OR_RETURN(value)                                                 \
+#define _VALID_OR_RETURN(_return)                                              \
   do {                                                                         \
-    lldbassert(!HasFatalErrors());                                             \
     if (HasFatalErrors()) {                                                    \
-      return (value);                                                          \
+      DiagnosticManager diagnostic_manager;                                    \
+      CopyDiagnostics(diagnostic_manager);                                    \
+      llvm::errs() << diagnostic_manager.GetString();                          \
```

Comment:
I'm worried that dumping this to an error channel will just cause this to get lost.  For instance, when lldb is used in an IDE stderr tends to be the system console.  It might be better to make it go to one of lldb's log channels (as well if not instead.)  Also, VALID_OR_RETURN can get called a lot of times as we back out of trying to do something with an AST context that had errors.  So you are likely to really spam whatever output channel you are sending this to.  It would be good to try this out and make sure that this doesn't get really obnoxious.  If it does we might want to have some way to emit the diagnostics once per SwiftASTContext.